### ISSUE TITLE
Prevent completing-read sorting

### ISSUE THREAD

**Issue by hmelman** [AnselmC/le-thesaurus.el#17]
This is very nice but when I first used it with [vertico](https://github.com/minad/vertico) enabled, the list of completion candidates was sorted by shortest words first (as I have vertico configured to sort candidates).  I see you do sort the candidates by similarity which is great. I think the proper way to tell a completion framework to not re-sort the candidates is to use a completion table with `metadata` specified for `display-sort-function`.  See "Programmed Completion" in the elisp manual.  For these candidates you might also benefit from specifying a `group-function` that groups the different "definition" values.

**Comment by AnselmC** [comment]
Hey Howard, thanks for this. My elisp skills are still in its infancy so I wasn't aware of this. I had noticed the same thing using selectrum but wasn't sure how to solve it so thank you for the pointers!
I added support to your suggestions (sorting & grouping) in this PR: https://github.com/AnselmC/le-thesaurus.el/pull/18
I'd be happy to receive review comments if you find the time - I may merge this soon though, so retrospectively is fine as well :)

<img width="459" alt="Screen Shot 2022-05-16 at 9 57 41 AM" src="https://user-images.githubusercontent.com/22545864/168545949-ca970537-11e2-413b-bf3c-e3fe4eb1bd84.png">
